src/game/network/gameinfo.cpp

```diff
@@ -13,7 +13,869 @@
+        Set_Slot_Pointer(i, &m_gameSlot[i]);
+    }
+}
+
+void SkirmishGameInfo::Xfer_Snapshot(Xfer *xfer)
+{
+    unsigned char version = 4;
+    xfer->xferVersion(&version, 4);
+    xfer->xferInt(&m_isPlayerPreorder);
+    xfer->xferInt(&m_crcInterval);
+    xfer->xferBool(&m_inGame);
+    xfer->xferBool(&m_inProgress);
+    xfer->xferBool(&m_unk);
+    xfer->xferInt(&m_gameID);
+    int slot_count = MAX_SLOTS;
+    xfer->xferInt(&slot_count);
+    captainslog_dbgassert(slot_count == MAX_SLOTS, "MAX_SLOTS changed, need to change version.");
+
+    for (int i = 0; i < MAX_SLOTS; i++) {
+        SlotState state = m_slot[i]->Get_State();
+        xfer->xferInt(reinterpret_cast<int *>(&state));
+        Utf16String name = m_slot[i]->Get_Name();
+
+        if (version >= 2) {
+            xfer->xferUnicodeString(&name);
+        }
+
+        bool accepted = m_slot[i]->Is_Accepted();
+        xfer->xferBool(&accepted);
+        bool unk = m_slot[i]->Get_Unk();
+        xfer->xferBool(&unk);
+        m_slot[i]->Set_Unk(unk);
+        int color = m_slot[i]->Get_Color();
+        xfer->xferInt(&color);
+        int start_pos = m_slot[i]->Get_Start_Pos();
+        xfer->xferInt(&start_pos);
+        int player_template = m_slot[i]->Get_Player_Template();
+        xfer->xferInt(&player_template);
+        int team = m_slot[i]->Get_Team_Number();
+        xfer->xferInt(&team);
+        int original_color = m_slot[i]->Get_Original_Color();
+        xfer->xferInt(&original_color);
+        int original_start_pos = m_slot[i]->Get_Original_Start_Pos();
+        xfer->xferInt(&original_start_pos);
+        int original_player_template = m_slot[i]->Get_Original_Player_Template();
+        xfer->xferInt(&original_player_template);
+
+        if (xfer->Get_Mode() == XFER_LOAD) {
+            m_slot[i]->Set_State(state, name, 0);
+
+            if (accepted) {
+                m_slot[i]->Set_Accepted();
+            }
+
+            m_slot[i]->Set_Player_Template(original_player_template);
+            m_slot[i]->Set_Start_Pos(original_start_pos);
+            m_slot[i]->Set_Color(original_color);
+            m_slot[i]->Save_Off_Original_Info();
+            m_slot[i]->Set_Team_Number(team);
+            m_slot[i]->Set_Color(color);
+            m_slot[i]->Set_Start_Pos(start_pos);
+            m_slot[i]->Set_Player_Template(player_template);
+        }
+    }
+
+    xfer->xferUnsignedInt(&m_localIP);
+    xfer->xferMapName(&m_mapName);
+    xfer->xferUnsignedInt(&m_mapCRC);
+    xfer->xferUnsignedInt(&m_mapSize);
+    xfer->xferInt(&m_mapContentsMask);
+    xfer->xferInt(&m_seed);
+
+    if (version < 3) {
+        if (xfer->Get_Mode() == XFER_LOAD) {
+            m_superweaponRestriction = 0;
+            m_money = g_theWriteableGlobalData->m_defaultStartingCash;
+        }
+    } else {
+        xfer->xferUnsignedShort(&m_superweaponRestriction);
+
+        if (version == 3) {
+            bool b;
+            xfer->xferBool(&b);
```

Comment:
What is this? Something missing here?

Reply:
This is correct. Don't ask me what its doing but its correct.